geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/KryoLazyDensityCoprocessor.scala

```diff
@@ -69,85 +69,115 @@ class KryoLazyDensityCoprocessor extends KryoLazyDensityService with Coprocessor
   def getService: Service = this
 
   def getDensity(controller: RpcController, request: KryoLazyDensityProto.DensityRequest, done: RpcCallback[KryoLazyDensityProto.DensityResponse]) {
-    val byteFilterArray = request.getByteFilter.toByteArray
     var response : DensityResponse = null
     var scanner : InternalScanner = null
+    var filterList : FilterList = new FilterList()
+
     try {
-      val scan = new Scan
-      val options: Map[String, String] = deserializeOptions(byteFilterArray)
+      val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
       val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
+      var scanList : List[Scan] = List[Scan]()
+
+      options(RANGES_OPT).split(",").foreach(range => {
+        val scanInfo = range.split("\\|")
+        scanList ::= new Scan(Base64.decode(scanInfo(0)), Base64.decode(scanInfo(1)))
+      })
+
+      if (options.containsKey(FILTER_OPT)) {
+        filterList = FilterList.parseFrom(Base64.decode(options(FILTER_OPT)))
+      }
+
       val serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId)
       val densityResult: DensityResult = this.init(options, sft)
-      scanner = env.getRegion.getScanner(scan)
-      val results = new java.util.ArrayList[Cell]
-      var hasMore = false
-      do {
-        hasMore = scanner.next(results)
-        for (cell <- results) {
-          val sf = serializer.deserialize(cell.getValueArray, cell.getValueOffset, cell.getValueLength)
-          aggregateResult(sf, densityResult)
-        }
-        results.clear()
-      } while (hasMore)
+
+      scanList.foreach(scan => {
+        scan.setFilter(filterList)
+        scanner = env.getRegion.getScanner(scan)
```

Comment:
let's add a TODO here to explore the MultiRangeFilter, that seemed to be much faster than lots of single scans.